sdk/nodejs/query/interfaces.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//
+// NOTE: We don't need `Disposable` or `Enumerator#reset()` yet, but it's worth noting here that we
+// didn't include them, and this causes us to diverge slightly from the "canonical" `Enumerable`
+// model.
+//
+
+export interface Enumerable<T> {
```

Comment:
If we do end up sticking with these interfaces... If I have an an `Enumerable<T>` object, how do I get its `Enumerator<T>` if I wanted to iterate over it manually with an enumerator? I see for `EnumerablePromise<T>` I can presumably get it from the resolved promise.

Reply:
We had not planned to support this. But, with the move to async iterators, we get it incidentally.